### PR TITLE
add new state for external software download clicked and emit the corresponding event

### PR DESCRIPTION
**Pull request by ibrahimahmed443**

### CODE REVIEW

File: edx_proctoring/models.py [hunk 1]
```diff
@@ -127,6 +127,10 @@ class ProctoredExamStudentAttemptStatus(object):
     # in the process of completing the exam
     started = 'started'
 
+    # the student has clicked on the external
+    # software download link
+    download_software_clicked = 'download_software_clicked'
```
Comment (afzaledx) [inline edx_proctoring/models.py]: Please add this new state in the `is_incomplete_status` method as well.

File: edx_proctoring/templates/proctored_exam/instructions.html [hunk 2]
```diff
@@ -107,5 +107,20 @@
...
+      var url = $('.instructions').data('exam-started-poll-url');
+      var action = $(this).data('action');
+      $.ajax({
+        url: url,
+        type: 'PUT',
+        data: {
+            action: action
+        },
+        success: function(data) {
+          console.log(data);
```
Comment (afzaledx) [inline edx_proctoring/templates/proctored_exam/instructions.html]: We need to open the URL in a new tab.

File: edx_proctoring/templates/proctored_exam/instructions.html [hunk 3]
```diff
@@ -26,7 +26,7 @@
       {% endblocktrans %}
     </p>
     <p>
-      <span><a href="{{software_download_url}}" target="_blank">Start System Check</a></span>
+      <span><a href="{{software_download_url}}" id="software_download_link" data-action="click_download_software" target="_blank">Start System Check</a></span>
```
Comment (afzaledx) [inline edx_proctoring/templates/proctored_exam/instructions.html]: Remove the href from here and redirect to the URL once the ajax call returns successfully.

**Comment by afzaledx** [comment]
:+1:

**Comment by chrisndodge** [comment]
@afzaledx @ibrahimahmed443 thanks. I just added some more unit tests. Merging.

**Pull request merged**